juju/juju
# Model credential validation update to support all model types.

## Description of change

Juju validates a cloud credential for a given model. 

However, CAAS models need to be treated differently as kubernetes provider does not allow the same visibility as traditional Juju providers. For example, we cannot compare the list of instances with the list of Juju-known machines.

This PR ensures that even CAAS models have their credential validated.

## QA steps

1. bootstrap and add caas model
2. call 'update-credential' command on the credential that the model in (1) uses

Repeat for traditional, IAAS model.

Expected outcome - no errors and operations succeed.
